source/plugins/ruby/kubelet_utils.rb

```diff
@@ -41,6 +41,111 @@ def get_node_capacity
+          @log.error "Error in get_node_allocatable::systemReserved_cpu: #{errorStr}"
+          systemReserved_cpu = "0"
+          ApplicationInsightsUtility.sendExceptionTelemetry("Error in get_node_allocatable::kubereserved_cpu: #{errorStr}")
+        end 
+        begin
+           systemReserved_memory = JSON.parse(allocatable_response.body)["kubeletconfig"]["systemReserved"]["memory"]
+           if systemReserved_memory.nil? || systemReserved_memory == ""
+            systemReserved_memory = "0"
+           end
+           @log.info "get_node_allocatable::systemReserved_memory #{systemReserved_memory}"
+        rescue => errorStr
+           @log.error "Error in get_node_allocatable::systemReserved_memory: #{errorStr}"
+           systemReserved_memory = "0"
+           ApplicationInsightsUtility.sendExceptionTelemetry("Error in get_node_allocatable::kubereserved_cpu: #{errorStr}")
+        end 
+
+        begin
+          evictionHard_memory = JSON.parse(allocatable_response.body)["kubeletconfig"]["evictionHard"]["memory.available"]
+          if evictionHard_memory.nil? || evictionHard_memory == ""
+            evictionHard_memory = "0"
+          end
+          @log.info "get_node_allocatable::evictionHard_memory #{evictionHard_memory}"
+        rescue => errorStr
+          @log.error "Error in get_node_allocatable::evictionHard_memory: #{errorStr}"
+          evictionHard_memory = "0"
+          ApplicationInsightsUtility.sendExceptionTelemetry("Error in get_node_allocatable::kubereserved_cpu: #{errorStr}")
+        end 
+
+        # do calculation in nanocore since that's what KubernetesApiClient.getMetricNumericValue expects
+        cpu_capacity_number = cpu_capacity.to_i * 1000.0 ** 2
+        # subtract to get allocatable. Formula : Allocatable = Capacity - ( kube reserved + system reserved + eviction threshold )
+        # https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/#node-allocatable
+        # cpu_allocatable  = cpu_capacity_number - ( kubereserved_cpu.tr('^0-9', '').to_i + systemReserved_cpu.tr('^0-9', '').to_i );
+        cpu_allocatable  = cpu_capacity_number - (KubernetesApiClient.getMetricNumericValue("cpu", kubereserved_cpu) + KubernetesApiClient.getMetricNumericValue("cpu", systemReserved_cpu))
+        cpu_allocatable = cpu_allocatable / (1000.0 ** 2)
+        @log.info "CPU Allocatable #{cpu_allocatable}"
+
+        memory_allocatable = memory_capacity - (KubernetesApiClient.getMetricNumericValue("memory", kubereserved_memory) + KubernetesApiClient.getMetricNumericValue("memory", systemReserved_memory) + KubernetesApiClient.getMetricNumericValue("memory", evictionHard_memory))
+        @log.info "Memory Allocatable #{memory_allocatable}"
+
+        # cpu_allocatable = BigDecimal(cpu_allocatable.to_s).to_f
+        # memory_allocatable = BigDecimal(memory_allocatable.to_s).to_f
+
+        return [cpu_allocatable, memory_allocatable]
```

Comment:
Yeah, this will be an issue in hybrid as people can customize the node cpu in a way they want. Currently there is no way in managed kubernetes to set aside reserved cpus. But it could be part of future node customization that we need to be aware..
Since MDM metrics doesn;t apply to un-managed now, this seems ok?

Reply:
We do support MDM metrics for hybrid via azure arc. If the reserved-cpus takes precedence, is there any reason not considering that?

Reply:
I see, updating the code to take this into account now that I know its being used in azure arc.